app/models/upload.rb

```diff
@@ -6,6 +6,11 @@ class Upload < ApplicationRecord
   belongs_to :stream, touch: true
   has_one :organization, through: :stream
   has_many :marc_records, dependent: :delete_all
+  validates :url, presence: true, if: proc { |upload| upload.files.blank? }
+  validates :files, presence: true, if: proc { |upload| upload.url.blank? }
+  validate :valid_url, if: proc { |upload| upload.url.present? }
+
+  after_create :attach_file_from_url
```

Comment:
I'm doing this in an `after_create`, which begs the question what editing/updating an upload should do.  I can change this to `after_save` and allow URLs to be re-uploaded, but not sure if that's the desired behavior or not (and if so, would we want to check if the URL had changed before re-downloading so that a change to the name doesn't trigger re-downloading the file).